### DISCUSSION TITLE
Contributing guidance

### DISCUSSION THREAD

**Discussion by Yash12Bansal** [robolectric/robolectric#7140] [Q&A] [answered]
I am new to this organization. I have read about this organization from various sources but want to get deeper insights into the work. So please guide me.

**Comment by utzcoz** [comment, marked as answer]
You can read http://robolectric.org/contributing/ firstly to setup Robolectric development environment locally and run current tests on your development machine. And then reading http://robolectric.org/extending/ to learn what Shadow(one of most important concept of Robolectric) works for on Robolectric. I also recommend to read other parts of robolectric.org to learn other user guides, because Robolectric provides what user can use. If you understand what Robolectric can do, please focus on one topic of you have interested in, e.g. Shadows, third-party integrations, code quality, CI.

> So please guide me.

It's recommended to follow contributing guides to setup Robolectric step by step, and try to fix some existing "help wanted" issues(https://github.com/robolectric/robolectric/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) for Robolectric with your PRs. It's very welcome to send PR for problems of Robolectric we didn't notice after you are familiar with Robolectric. If you encounter problem when learning and improving Robolectric, please show your questions and problems, we can work together to overcome those problems. It's also welcome to show which part do you need for contributing guide, and we can discuss whether to add it for you and later contributors.

Contribution of any kind is greatly appreciated.
